### ISSUE TITLE
Microsoft Edge 622% slower with Blazor websites

### ISSUE THREAD

**Issue by Pinox** [dotnet/aspnetcore#48754]
Hi there,

Just want to know if I'm the only one experiencing very slow MS Edge browser performance on Blazor (WASM) websites.
I have noticed that using MS Edge during development takes forever to load the initial assets compared to other browsers.

I have decided for comparison to use my favorite Blazor UI Library - Mudblazor main website for comparison.

www.mudblazor.com

Initial load on MS Edge :  14.61s
![image](https://github.com/dotnet/aspnetcore/assets/5297255/8a2d5f58-e580-4224-b57e-173f542e23f6)

Initial load on Google Chrome :  2.94s
![image](https://github.com/dotnet/aspnetcore/assets/5297255/1cce7a7f-bbd4-4ed1-95b7-fbcc79c9b324)

Initial load on Opera:  2.35s
![image](https://github.com/dotnet/aspnetcore/assets/5297255/f3bba822-0336-4cc0-8631-1a5d1899458a)

As you can see MS edge is **622%** slower than Opera  (I used the networks tab in the browser dev tools)

I also disabled all my extensions in my MS Edge browser as I can't seem to find the reason for this poor performance from Edge on Blazor websites.

I then tried to run the same test on MS Edge within Windows Sandbox that is effectively a clean install.
Initial load on MS Edge inside Windows Sandbox :  3.11s
![image](https://github.com/dotnet/aspnetcore/assets/5297255/b73bb16f-470b-4f0d-9316-57816ee7ddbd)

Just to make sure this is not a faulty installation of Edge on my desktop PC I cross checked the results with a gaming laptop and I have the same results on another Windows 11 machine.

So there seems to be a difference between a clean install of MS Edge within Windows Sandbox and my current Windows 11 MS edge browser on 2 independent PC's.  Is there a way to fix this performance issue or a way to clean install Edge on Windows 11 or is this a known issue for Edge on Windows 11 ?

Thanks in advance.

**Comment by SteveSandersonMS** [comment]
cc @danroth27 for the latest info on this.

**Comment by danroth27** [comment]
@pinox Could you please submit this feedback directly to the Edge folks using the Edge Send Feedback option under the Help and Feedback menu? There's very little we can do about the impact of Edge Enhanced Security Mode on our end, so this feedback needs to go to them directly.

**Comment by Pinox** [comment]
Thanks @danroth27. Will do.

**Comment by danroth27** [comment]
@Pinox Thanks! We're also following up with the Edge folks on our side.

**Comment by Pinox** [comment]
Just another note on this for record purposes, it obviously affects all websites and not just Blazor websites.  Just tested this on facebook marketplace and now the "download heavy" website are performing well again.  So all facebook users on MS Edge will have the same severe degraded experience when the "Strict Mode" is selected in enhanced security on Edge.

**Comment by danroth27** [comment]
@Pinox Do you know how you ended up with Strict mode enabled? Did you turn that on yourself? Or perhaps it was rolled out to your machine via an organizational policy?

**Comment by Pinox** [comment]
@danroth27  Dan I think I might have changed it in the past, not 100% sure.  Had a scenario in the past (about 2 years ago) where my google search result were injected with redirected traffic (Spigots) so I ended up vetting and minimizing my extensions to minimize the attack surface and installed Malwarebytes to get rid of the browser hijackers. 

Looking at the Edge setting ( Strict Mode), it looks innocent enough to use as I will pick up quickly if my regular websites do not work. When MS changed to the chromium engine ,  I moved from Chrome to Edge as my default browser.

However because of poor performance of Edge in development (loading times) , chrome was my default for development.  In the back of head I knew sometime was wrong with Edge but did not suspect this setting as everything was working fine with all websites, they were all just very slow in loading the initial pages.  

As I'm currently converting a native windows app to Blazor I was concerned about this lag in WASM sites loading 
( using www.mudblazor.com as a test ) in Edge therefor me starting to ask questions about this.

I do think MS should clearly highlight that this "Strict Mode" should not be used in development and also that it is 6 times slower in loading website assets.  With many big brand websites that's very big in download size this setting ( Strict mode) is "catastrophic" in my opinion.   As stated above facebook markerplace suddenly feels light and responsive again.  I think most people will never select "Strict mode" if the performance of this feature is pointed out to them.
